compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -10533,16 +10531,20 @@ OMR::Z::TreeEvaluator::arraycmpHelper(TR::Node *node,
       TR::Register *baseReg1Temp = NULL;
       TR::Register *baseReg2Temp = NULL;
 
+
+      // if any of the to be generated CLC instructions will need a large disp or the reference offsets are not equal, generate an LA up front.
+      // if the base registers are equal, the arrays are considered equal therefore the displacements must be equal.
```

Comment:
Can you elaborate, how if base registers are equal, displacement would always be equal ?

Reply:
Sure, I'll write a more clear explanation.

Reply:
Update is making it more confusing. Comment should convey the following insights more clearly.
1. If the source1 and source2 array addresses are same, we do not need to compare array elements.
2. If the displacement is large or source1 and source2 memref's displacement are not equal, we force the LA instruction to materialize the src1 and src2 array address into register. 

> if the base registers are equal, the arrays are considered equal therefore the displacements must be equal.

Above line in the comment (Especially as the comment appears before you check the offset), suggests that the code is making an assumption that if the base registers for both arrays are equal, displacement should be equal. While your code does handle the case where if the displacements are unequal, you materialize it into the register.

At first I though, if we decide to generate arraycmp for comparing the two sections of same arrays (which would have same base address for array, but difference offset), it will simply compare the base address and return true. Further looking the code, it became more clear.

Reply:
I added more context. I hope it is clear this time.